airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/pipeline.py

```diff
@@ -54,7 +54,7 @@ async def run_connector_test_pipeline(context: ConnectorTestContext, semaphore:
 
     all_steps_to_run += get_test_steps(context)
 
-    if not context.code_tests_only:
+    if not context.code_tests_only and not context.metadata_yaml_only_change:
```

Comment:
can you also create another method in ConnectorTestContext called test_only_change that returns true if the only modified files are in the unit_tests and integration_tests.

And here we should only add static_analysis_steps_to_run if the current test context has modified files not including metadata.yaml, unit_tests, and integration_tests